### ISSUE TITLE
Waypoints of ALC sometimes not shown on OSM maps

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13675]
### Describe your problem!

A user of 2022.11.13 reported, that he likes the new feature, however has problems, that waypoints of ALC vanish or are not visible when mapping a list (also) containing ALC.

Its not clearly reproducible (but as I also had these symptoms before) I am quite sure there is a bug in there.

### How to reproduce?

The user clearly reported, that it works on Google maps and the problem is only existing on OSM/Offline maps.

### Reproducible

Unclear

### c:geo Version

2022.11.13

**Comment by Lineflyer** [comment]
Addition:
When I had this problem I somehow had the theory, that the "waypoint show limit" for the map treats ALC waypoints different than regular cache waypoints on OSM maps. 
But thats mainly a "wild guess".

**Comment by Lineflyer** [comment]
I could kind of reproduce it today on a OSM offline map in live mode (no mapped list).

Precondition is something like:
- Have some normal caches around
- Have some stored ALC around (maybe some of them found...no idea, whether this is important)
- In my case filtering of found caches was active (no idea, whether this is important)
- Open/close popup repeatedly to enforce map update (This is probably only one means to trigger the bug, moving the map around produces the same but more difficult to show)

Video attached:
https://user-images.githubusercontent.com/949669/202897074-f8af41ad-1048-4ee3-8419-4c8e36e20c88.mp4

Used cache is https://adventurelab.page.link/wGUP

**Comment by Lineflyer** [comment]
I was also able to reproduce the above mentioned procedure on an emulator.
At the point where the waypoint is not shown, I do see repetitive debug messages like this (without any user interaction on the map) which repeat endless until I trigger an action to make the waypoint show up again.

So it seems related to the bug.

```
2022-11-20 11:32:27.529  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:27.780  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:28.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:28.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:28.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:29.031  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:29.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:29.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:30.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:30.279  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 1 line
2022-11-20 11:32:30.531  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:30.780  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:31.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:31.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:31.780  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:32.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:32.529  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:32.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:33.029  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:33.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:33.780  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:34.035  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:34.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:34.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:35.029  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:35.279  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 1 line
2022-11-20 11:32:35.529  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:35.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:36.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:36.529  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:36.780  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:37.030  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:37.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:37.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:38.029  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:38.530  7793-8024  chatty       cgeo.geocaching  I  uid=10154(cgeo.geocaching) RxNewThreadSche identical 2 lines
2022-11-20 11:32:38.779  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}
2022-11-20 11:32:39.029  7793-8024  cgeo         cgeo.geocaching  D  [RxNewThreadScheduler-11] Waypoint 1 for id 0 not added, geoEntries: 53 {AbstractCachesOverlay.addItem:220/WaypointsOverlay.showWaypoints:63/CachesBundle.handleWaypoints:345/AbstractCachesOverlay.updateTitle:282/StoredCachesOverlay$LoadTimerAction.run:75/Scheduler$PeriodicDirectTask.run:596/Scheduler$Worker$PeriodicTask.run:543/ScheduledRunnable.run:65}

```

**Comment by Lineflyer** [comment]
Anyone able to dig this one down based on the reproduction method and logs mentioned above?

**Comment by moving-bits** [comment]
We do have a couple of problems with caches / waypoints sometimes not being shown on OSM map, but working on GMv2.

My guess it's some timing problem (eg: new refresh of map items already triggered, although not all items are already having been drawn). GMv2 uses a queue for painting map items - maybe that's the reason why GMv2 is not affected. 
The discussion that started in https://github.com/cgeo/cgeo/pull/13911#issuecomment-1411813085 is about introducing something similar for all our map types.

**Comment by eddiemuc** [comment]
> GMv2 uses a queue for painting map items - maybe that's the reason why GMv2 is not affected.

A bit off-topic, but the current queueing algorithm used in GMv2 is not thread-safe! It is bound to produce wrong results when large groups of objects are fastly removed and added again

**Comment by moving-bits** [comment]
should no longer happen since #14255 (included in release 2023.05.15)

@Lineflyer 
Can you cross-check (as time permits)?

**Comment by Lineflyer** [comment]
I was not using c:geo recently, but some weekends ago I did some lab + multi combination and did not run into this issue. So lets close for now. I will pay attention, whether it shows up again.
Thanks for fixing!
